src/components/input/input.js

```diff
@@ -191,6 +242,10 @@ function inputTextareaDirective($mdUtil, $window, $mdAria) {
     }
     containerCtrl.input = element;
 
+    // Add an error spacer div after our input to provide space for the char counter and any ng-messages
+    var errorsSpacer = angular.element('<div class="md-errors-spacer">');
```

Comment:
I tried a pure CSS solution, but ran into issues with the fact that I needed to position the `md-char-counter` at that exact spot. With pure CSS, the counter would move if the input had more than 1 message which made it bigger.

It's possible there is another way, but I was in a bit of a rush and I spent way too much time trying to find a solution before thinking of this and implementing it in about 15 minutes.